### PR TITLE
chore: add RTL CSS Checker to autofix

### PR DESCRIPTION
**Pull request by userquin**
The file paths with errors using absolute path, clicking using relative paths in WebStorm terminal doesn't work (VSCode click works with absolute and relative paths).

This PR includes a fix at `DownloadAnalitics.vue` for a `text-right` style, `preset-wind4` adding some rules with higher priority and the RTL preset dind't detect those cases, solved using shorcuts and a hack.

This afternoon I'll try to add the fix: https://streamable.com/fr498h

### CODE REVIEW

File: uno-preset-rtl.ts [hunk 1]
```diff
@@ -78,74 +87,108 @@ function handlerBorderSize([, a = '', b = '1']: string[]): CSSEntries | undefine
...
           if (!replacement) return undefined
-          warnOnce(
-            `[RTL] Avoid using 'rounded-${direction}'. Use 'rounded-${replacement}' instead.`,
-            `rounded-${direction}`,
-          )
+
+          const fullClass = context.rawSelector || match
+          const suggestedBase = match.replace(direction!, replacement)
+          const suggestedClass = fullClass.replace(match, suggestedBase)
+
```
[the remark this answers is not in the document]
Comment (userquin) [reply uno-preset-rtl.ts:169]: same edge case with rounded and border, rules and test fixed

File: uno-preset-rtl.ts [hunk 2]
```diff
@@ -78,74 +87,108 @@ function handlerBorderSize([, a = '', b = '1']: string[]): CSSEntries | undefine
...
           const replacement = direction === 'l' ? 'is' : 'ie'
-          warnOnce(
-            `[RTL] Avoid using 'border-${direction}'. Use 'border-${replacement}' instead.`,
-            `border-${direction}`,
-          )
+
+          const fullClass = context.rawSelector || match
+          const suggestedBase = match.replace(direction!, replacement)
+          const suggestedClass = fullClass.replace(match, suggestedBase)
+
```
[the remark this answers is not in the document]
Comment (userquin) [reply uno-preset-rtl.ts:183]: fixed

File: uno-preset-rtl.ts [hunk 3]
```diff
@@ -78,74 +87,108 @@ function handlerBorderSize([, a = '', b = '1']: string[]): CSSEntries | undefine
...
+          if (checker) {
+            checker(`avoid using '${fullClass}'. Use '${suggestedClass}' instead.`, fullClass)
+          } else {
+            warnOnce(
+              `[RTL] Avoid using '${fullClass}'. Use '${suggestedClass}' instead.`,
+              fullClass,
+            )
+          }
```
[the remark this answers is not in the document]
Comment (userquin) [reply uno-preset-rtl.ts]: fixed, added also some tests

**Comment by userquin** [comment]
~~DONT MERGE YET: I have a regresion with `text-(left|right)`, the test snapshot not replacing them with `text-(start|end)` 🤔~~

fixed

File: uno-preset-rtl.ts [hunk 4]
```diff
@@ -38,18 +40,25 @@ const directionMap: Record<string, string[]> = {
...
+    const suggestedBase = match.replace(`${prefix}${direction!}`, `${prefix}${replacement}`)
+    const suggestedClass = fullClass.replace(match, suggestedBase)
+
+    if (checker) {
+      checker(`avoid using '${fullClass}', use '${suggestedClass}' instead.`, fullClass)
+    } else {
+      warnOnce(`[RTL] Avoid using '${fullClass}'. Use '${suggestedClass}' instead.`, fullClass)
+    }
     return matcher([match, replacement, size], context)
   }
```
[the remark this answers is not in the document]
Comment (userquin) [reply uno-preset-rtl.ts:69]: Fixed. By the way, didn't you notice that `text-left`, `text-right`, and `hover:text-right` were still using their original values? That's a problem with the shortcuts in InoCSS's preset-wind4.

File: uno-preset-rtl.ts [hunk 5]
```diff
@@ -78,74 +93,96 @@ function handlerBorderSize([, a = '', b = '1']: string[]): CSSEntries | undefine
...
           )
-          return { 'text-align': replacement }
+
+          reportWarning(originalClass, suggestedClass, checker)
+
+          // Return a cssvar with the warning message to satisfy UnoCSS
+          // and avoid "unmatched utility" warning.
+          return {
+            [`--x-rtl-${direction!}`]: `"${originalClass} -> ${suggestedClass}"`,
+          }
```
[the remark this answers is not in the document]
Comment (userquin) [reply uno-preset-rtl.ts:151]: In production won't have any text-left/right since this PR will prevent those css styles, check the title and the autofix workflow, it will run `scripts/rtl-checker.ts`

File: uno-preset-rtl.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (userquin) [reply uno-preset-rtl.ts:69]: preset-wind4 defines text-* using shorcuts or something with higher priority:
```ts
// https://github.com/unocss/unocss/blob/main/packages-presets/preset-wind4/src/rules/align.ts
const textAlignValues = ['center', 'left', 'right', 'justify', 'start', 'end']

export const textAligns: Rule<Theme>[] = [
  ...textAlignValues
    .map(v => [`text-${v}`, { 'text-align': v }] as Rule<Theme>),
  ...[
    ...globalKeywords,
    ...textAlignValues,
  ].map(v => [`text-align-${v}`, { 'text-align': v }] as Rule<Theme>),
]
```

**Comment by userquin** [comment]
PR ready to review.

File: uno-preset-rtl.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (userquin) [reply uno-preset-rtl.ts:151]: Did my explanation not resolve the potential issue? Why haven't you resolved it?

**Review by danielroe — APPROVED** [review]
